fdbclient/BackupContainer.h

```diff
@@ -108,6 +108,12 @@ struct RangeFile {
 	std::string fileName;
 	int64_t fileSize;
 
+	RangeFile() {}
```

Comment:
It's not clear to me why these were added.  They appear to be the same as the default initializers would be, which were already being used with the initializer syntax `{}`.

Reply:
The compiler generates errors for simple declaration like "`RangeFile file;`" if this constructor is no given, because I added another constructor taking 4 parameters below.

Reply:
I'm not just talking about the default, I mean all three constructors.  The 4-parameter constructor seems to be exactly what the default struct initializer list constructor would do, and the copy constructor does what the default would do.  In the absence of those, the default constructor is also unnecessary.  Previously, range files were being constructed with initializer lists.  Without any of those constructors, you can still things like:

```
RangeFile f;
RangeFile g = {version, blockSize, name, fileSize};
vector<RangeFile> v;
v.push_back({version, blockSize, name, fileSize});
```

Reply:
I'm not terribly against having them there, I just couldn't see why they were needed.  RangeFile is a just POD struct so such definitions are not necessary.

Reply:
I see your point now. We don't need these explicit constructors and the default can work just fine.

Reply:
I'm removing these constructors in #3036.